backend/metadata_writer/src/metadata_writer.py

```diff
@@ -116,6 +118,8 @@ def is_tfx_pod(pod) -> bool:
     main_container = main_containers[0]
     return main_container.command and main_container.command[-1].endswith('tfx/orchestration/kubeflow/container_entrypoint.py')
 
+def is_kfp_v2_pod(pod) -> bool:
+    return pod.metadata.labels[KFP_V2_COMPONENT_ANNOTATION_KEY] == KFP_V2_COMPONENT_ANNOTATION_VALUE
```

Comment:
This might fail if the label does not exist. Better try
```
return pod.metadata.labels.get(KFP_V2_COMPONENT_ANNOTATION_KEY) == KFP_V2_COMPONENT_ANNOTATION_VALUE
```

Reply:
Thanks @Ark-kun! That's very helpful